realm/realm-library/src/syncIntegrationTest/java/io/realm/BaseIntegrationTest.java

```
@@ -0,0 +1,102 @@
+        } catch (Exception e) {
+            // Throwing an exception from this method will crash JUnit. Instead just log it.
+            // If this setup method fails, all unit tests in the class extending it will most likely fail as well.
+            Log.e(HttpUtils.TAG, "Could not start Sync Server: " + Util.getStackTrace(e));
+        }
+    }
+
+    @AfterClass
+    public static void tearDown () throws Exception {
+        try {
+            HttpUtils.stopSyncServer();
+        } catch (Exception e) {
+            Log.e(HttpUtils.TAG, "Failed to stop Sync Server" + Util.getStackTrace(e));
+        }
+    }
+
+    @Before
+    public void setupTest() throws IOException {
+        // TODO We should implement a more consistent reset method for all of Sync that reset
+        // everything completely including deleting all files.
+        deleteRosFiles();
+        if (BaseRealm.applicationContext != null) {
+            // Realm was already initialized. Reset all internal state
+            // in order to be able fully re-initialize.
+            SyncManager.reset(); // Required for filesystem layout to be re-constructed.
+            BaseRealm.applicationContext = null; // Required for Realm.init() to work
+        }
+        Realm.init(InstrumentationRegistry.getContext());
+        originalLogLevel = RealmLog.getLevel();
+        RealmLog.setLevel(LogLevel.DEBUG);
+    }
+
+    @After
+    public void tearDownTest() throws IOException {
+        RealmLog.setLevel(originalLogLevel);
+    }
+
+
+    // Cleanup filesystem to make sure nothing lives for the next test.
+    // Failing to do so might lead to DIVERGENT_HISTORY errors being thrown if Realms from
+    // previous tests are being accessed.
+    private static void deleteRosFiles() throws IOException {
+        File rosFiles = new File(InstrumentationRegistry.getContext().getFilesDir(),"realm-object-server");
+        if (rosFiles.isDirectory()) {
+            deleteFile(rosFiles);
+        }
```

Then something is seriously wrong, but yes, I should probably throw an error in that case.